src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotator.java

```diff
@@ -188,23 +189,27 @@
      */
     public void onTraversalStart() {
         // get the list of all sample names from the variant VCF, if applicable
-        final  List<String> samples = getHeaderForVariants().getGenotypeSamples();
+        final List<String> samples = getHeaderForVariants().getGenotypeSamples();
         variantSamples = new IndexedSampleList(samples);
 
         annotatorEngine = new VariantAnnotatorEngine(makeVariantAnnotations(), dbsnp.dbsnp, comps, false, false);
-        annotatorEngine.addExpressions(expressionsToUse, resources, expressionAlleleConcordance );
+        annotatorEngine.addExpressions(expressionsToUse, resources, expressionAlleleConcordance);
 
         // setup the header fields
         // note that if any of the definitions conflict with our new ones, then we want to overwrite the old ones
-        final Set<VCFHeaderLine> hInfo = new HashSet<>();
-
+        Set<VCFHeaderLine> hInfo = new HashSet<>();
         hInfo.addAll(annotatorEngine.getVCFAnnotationDescriptions(false));
         hInfo.addAll(getHeaderForVariants().getMetaDataInInputOrder());
 
-        // TODO ask reviewer, VCFUtils.withUpdatedContigs is what GATK3 calls into, it isn't used anywhere in 4 though so should it be used here?
-        VCFHeader vcfHeader = new VCFHeader(hInfo, samples);
+        if (hasReference()) {
+            hInfo = VcfUtils.updateHeaderContigLines(
+                    hInfo,
+                    referenceArguments.getReferencePath(),
+                    getReferenceDictionary(),
```

Comment:
The previous implementation fell back on `getBestAvailableSequenceDictionary()` here -- was there any point to having that as a fallback, or was it pointless?

Reply:
I think its mostly pointless, but there are cases where not doing so would alter behavior (ie when a master sequence dictionary is provided but not a reference). So I'm going to restore the fallback for now. (Unifying the various patterns for output VCFHeader creation is another candidate for a refactoring project at some point).